### DISCUSSION TITLE
undefined symbol  ffi_type_uint32 version LIBFFI_BASE_7.0

### DISCUSSION THREAD

**Discussion by dyxer** [astra-toolbox/astra-toolbox#494] [Q&A]
Hi,
I download the source code of astra toolbox from https://github.com/astra-toolbox/astra-toolbox, then created a python envs and activated it using conda, installing astra by using follow command line.
cd build/linux
./autogen.sh   # when building a git version
./configure --with-cuda=/usr/local/cuda \
            --with-python \
            --with-install-type=module
make
make install

But when i using the created astra envs to  run fdk algorithm, i met a bug that  ImportError: /lib/x86_64-linux-gnu/libgobject-2.0.so.0: undefined symbol: ffi_type_uint32, version LIBFFI_BASE_7.0. The detail infor as follows: 
Traceback (most recent call last):
  File "......./FDKReconstruction.py", line 20, in <module>
    import astra
  File "/....../site-packages/astra_toolbox-2.1.0-py3.8-linux-x86_64.egg/astra/__init__.py", line 26, in <module>
    from . import matlab as m
  File "/....../site-packages/astra_toolbox-2.1.0-py3.8-linux-x86_64.egg/astra/matlab.py", line 38, in <module>
    from . import astra_c
ImportError: /lib/x86_64-linux-gnu/libgobject-2.0.so.0: undefined symbol: ffi_type_uint32, version LIBFFI_BASE_7.0

I had tried many advices from github,such as  export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libffi.so.7,update libffi reinstall astra and so on,but it dosen't work.Look forward to your reply and suggestions.

**Comment by wjp** [comment]
Hi,
While I haven't seen this specific error, I've seen similar situations where libraries in the conda environment would conflict with libraries in the main system. Could you run `ldd astra_c.*.so` and `ldd libastra.so` (from the directory where those files are installed), both when you have your conda environment activated and when the conda environment is deactivated entirely? That might give some more clues to what's happening.

One other odd thing I notice is that it's referring to astra_toolbox-2.1.0, while current git versions would be 2.2.0, so another possibility is there are two separate versions of astra installed at the same time and that's mixing things up.
